example/src/main/java/com/example/filters/MainActivity.java

```diff
@@ -55,36 +55,106 @@ private void bindDataToAdapter() {
         Runnable r = new Runnable() {
             public void run() {
                 Bitmap thumbImage = Bitmap.createScaledBitmap(BitmapFactory.decodeResource(context.getResources(), R.drawable.photo), 640, 640, false);
-                ThumbnailItem t1 = new ThumbnailItem();
-                ThumbnailItem t2 = new ThumbnailItem();
-                ThumbnailItem t3 = new ThumbnailItem();
-                ThumbnailItem t4 = new ThumbnailItem();
-                ThumbnailItem t5 = new ThumbnailItem();
-                ThumbnailItem t6 = new ThumbnailItem();
-
-                t1.image = thumbImage;
-                t2.image = thumbImage;
-                t3.image = thumbImage;
-                t4.image = thumbImage;
-                t5.image = thumbImage;
-                t6.image = thumbImage;
-                ThumbnailsManager.clearThumbs();
-                ThumbnailsManager.addThumb(t1); // Original Image
-
-                t2.filter = SampleFilters.getStarLitFilter();
-                ThumbnailsManager.addThumb(t2);
-
-                t3.filter = SampleFilters.getBlueMessFilter();
-                ThumbnailsManager.addThumb(t3);
 
-                t4.filter = SampleFilters.getAweStruckVibeFilter();
-                ThumbnailsManager.addThumb(t4);
-
-                t5.filter = SampleFilters.getLimeStutterFilter();
-                ThumbnailsManager.addThumb(t5);
-
-                t6.filter = SampleFilters.getNightWhisperFilter();
-                ThumbnailsManager.addThumb(t6);
+                ThumbnailItem originalImage = new ThumbnailItem();
```

Comment:
Had to redo the object instantiations in order to make the code better understanding. I have used the respective filter names for variables.